app/javascript/src/Common/components/SelectWithModal.jsx

```diff
@@ -0,0 +1,140 @@
+// @flow
+
+import * as React from 'react'
+
+import {
+  FormGroup,
+  Select,
+  SelectVariant
+} from '@patternfly/react-core'
+import {
+  toSelectOption,
+  toSelectOptionObject,
+  SelectOptionObject
+} from 'utilities/patternfly-utils'
+import { TableModal } from 'Common'
+
+import type { Record } from 'utilities/patternfly-utils'
+
+type Props<T: Record> = {
+  item: T | null,
+  items: T[],
+  onSelect: (T | null) => void,
+  isDisabled?: boolean,
+  label: string,
+  id: string,
+  name?: string,
+  helperText?: React.Node,
+  placeholderText?: string,
+  maxItems?: number,
+  header?: string,
+  footer?: string,
+  cells: { title: string, propName: string }[],
+  modalTitle: string,
+}
+
+const MAX_ITEMS = 20
+const HEADER = 'Most recently created'
+const FOOTER = 'View all'
+
+const SelectWithModal = <T: Record>({
+  item,
+  items,
+  onSelect,
+  isDisabled,
+  label,
+  id,
+  name,
+  helperText,
+  placeholderText,
+  maxItems = MAX_ITEMS,
+  header = HEADER,
+  footer = FOOTER,
+  cells,
+  modalTitle
+}: Props<T>) => {
+  const headerItem = { id: 'header', name: header, disabled: true, className: 'pf-c-select__menu-item--group-name' }
+  const footerItem = { id: 'foo', name: footer }
+
+  const [expanded, setExpanded] = React.useState(false)
+  const [modalOpen, setModalOpen] = React.useState(false)
+
+  const handleOnSelect = (_e, option: SelectOptionObject) => {
+    setExpanded(false)
+
+    if (option.id === footerItem.id) {
+      setModalOpen(true)
+    } else {
+      const selectedBackend = items.find(b => String(b.id) === option.id)
+
+      if (selectedBackend) {
+        onSelect(selectedBackend)
+      }
+    }
+  }
+
+  const getItems = items => [
+    headerItem,
+    ...items.slice(0, maxItems).map(i => ({ ...i, className: 'pf-c-select__menu-item-description' })),
+    footerItem
+  ]
+
+  // $FlowFixMe $FlowIssue It should not complain since Record.id has union "number | string"
+  const options = getItems(items).map(toSelectOption)
+
+  const handleOnFilter = (e) => {
+    const term = new RegExp(e.target.value, 'i')
+
+    const filteredRecords: T[] = term !== '' ? items.filter(b => term.test(b.name)) : items
+    let filteredItems = filteredRecords.length === 0
```

Comment:
what's the reason of using `let` here?

Reply:
None, I must have left it from an older version.